docusaurus/docs/iOS/components/message-list.md

```diff
@@ -1,3 +1,272 @@
+    open var channelController: ChatChannelController!
+
+    /// User search controller for suggestion users when typing in the composer.
+    open lazy var userSuggestionSearchController: ChatUserSearchController =
+        channelController.client.userSearchController()
+
+    /// The message list component responsible to render the messages.
+    open lazy var messageListVC: ChatMessageListVC = ChatMessageListVC()
+
+    /// Controller that handles the composer view.
+    open lazy var messageComposerVC = ComposerVC()
+
+    /// Flag to control if it is currently loading previous messages.
+    @Atomic private var loadingPreviousMessages: Bool = false
+
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Layout
+        messageListVC.view.translatesAutoresizingMaskIntoConstraints = false
+        addChildViewController(messageListVC, targetView: view)
+        NSLayoutConstraint.activate([
+            messageListVC.view.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            messageListVC.view.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            messageListVC.view.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor)
+        ])
+
+        messageComposerVC.view.translatesAutoresizingMaskIntoConstraints = false
+        addChildViewController(messageComposerVC, targetView: view)
+        NSLayoutConstraint.activate([
+            messageComposerVC.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            messageComposerVC.view.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            messageComposerVC.view.topAnchor.constraint(equalTo: messageListVC.view.bottomAnchor),
+            messageComposerVC.view.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+
+        // Setup
+        messageListVC.delegate = self
+        messageListVC.dataSource = self
+        messageListVC.client = channelController.client
+
+        messageComposerVC.channelController = channelController
+        messageComposerVC.userSearchController = userSuggestionSearchController
+
+        channelController.setDelegate(self)
+        channelController.synchronize( { [weak self] _ in
+            self?.messageComposerVC.updateContent()
+        })
+    }
+}
+```
+
+:::tip 
+In order to be easier to setup child view controllers you can add this extension to your application:
+
+```swift
+extension UIViewController {
+    func addChildViewController(_ child: UIViewController, targetView superview: UIView) {
+        child.willMove(toParent: self)
+        addChild(child)
+        superview.addSubview(child.view)
+        child.didMove(toParent: self)
+    }
+}
+```
+:::
+
+Next, we need to implement the `ChatMessageListVCDataSource` to connect the messages from the `ChannelController` to the `ChatMessageListVC`.
+```swift
+extension ChannelViewController: ChatMessageListVCDataSource {
```

Comment:
we should not ask the customer to add this boilerplate code, ideally you get this out of the box (ie. call 1 method, inheritance, ...)

Reply:
What do you mean by calling one method? 🤔 The only way I'm seeing to make this easier is to have a `ChatMessageListChannelDataSource` and a `ChatMessageListChannelDelegate` already implemented. The problem is that, if we do this, it makes sense for our `ChatChannelVC` to use them as well, which means, `ChatChannelVC` won't be any more a `ChatMessageListChannelDataSource` and `ChatMessageListChannelDelegate`, so this will cause a breaking change.

The same applies to the Thread.